File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    // gets input from the user for the text to grade
    string text = get_string("Text: ");

    // finds the number of letters in the text
    int letters = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            letters++;
        }
    }

    // find the number of words in the text
    int words = 1;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
    }

    // finds the number of sentences in the text
    int sentences = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }

    // uses the Coleman-Liau index formula to calculate the grade of the text
    float calculation = (0.0588 * letters / words * 100) - (0.296 * sentences / words * 100) - 15.8;
    int index = round(calculation);

    // outputs the grade of the text
    if (index < 1)
    {
        printf("Before Grade 1\n");
        return 0;
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
        return 0;
    }
    else
    {
        printf("Grade %i\n", index);
        return 0;
    }
}

File: sentimental-readability/readability.py
from cs50 import get_string

# Gets input from the user
text = get_string("Text: ")

# Creates all variable for the letters, words and sentences
letters = 0
words = 1
sentences = 0

# Goes through the whole text
for i in text:
    # If the current character is a letter of the alphabet increment letters
    if i.isalpha():
        letters += 1
    # If the current character is a space increment words
    elif i.isspace():
        words += 1
    # If the current character is a ponctuation increment sentences
    elif i == "." or i == "!" or i == "?":
        sentences += 1
# Calculate grade index
index = round((0.0588 * letters / words * 100) - (0.296 * sentences / words * 100) - 15.8)

# Prints what grade the text is
if index < 1:
    print("Before Grade 1")

elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")